error: redefinition of 'dim3' when including CUDA or HIP headers

**Bug summary**
After recent changes (possibly #1773) SYCL code using interop with CUDA or HIP no longer compiles due to the error:
```
error: redefinition of 'dim3'
struct __device_builtin__ dim3
                          ^
[...]/include/AdaptiveCpp/hipSYCL/pcuda/detail/dim3.hpp:16:8: note: previous definition is here
struct dim3 {
       ^
```
This happens in any file including `<sycl/sycl.hpp>` and HIP or CUDA runtime / library headers.

**To Reproduce**
Minimal example with the cuBLAS header:
```console
$ cat test.cpp 
#include <sycl/sycl.hpp>
#include <cublas_v2.h>
int main() {}

$ acpp test.cpp -I/usr/local/cuda/include
In file included from test.cpp:2:
In file included from /usr/local/cuda/include/cublas_v2.h:69:
In file included from /usr/local/cuda/include/cublas_api.h:74:
In file included from /usr/local/cuda/include/driver_types.h:61:
/usr/local/cuda/include/vector_types.h:426:27: error: redefinition of 'dim3'
struct __device_builtin__ dim3
                          ^
[...]/include/AdaptiveCpp/hipSYCL/pcuda/detail/dim3.hpp:16:8: note: previous definition is here
struct dim3 {
       ^
1 error generated.
```

I've also seen the same happen with `hip_runtime.h`.

**Expected behavior**
Code including `<sycl/sycl.hpp>` and HIP/CUDA headers compiles fine.

**Describe your setup**
* AdaptiveCpp commit f0616c95c57073e51c8f561ae8b7c3756cfca34a
* Ubuntu 22.04 with clang-15 from apt
* CUDA 12.9
```
git clone https://github.com/AdaptiveCpp/AdaptiveCpp.git
cd AdaptiveCpp
cmake -Bbuild -DCMAKE_INSTALL_PREFIX=$PWD/install -G Ninja
cmake --build build
cmake --install build

export PATH=$PWD/install/bin:$PATH
export LIBRARY_PATH=$PWD/install/lib:$LIBRARY_PATH
export LD_LIBRARY_PATH=$PWD/install/lib:$LD_LIBRARY_PATH

$ acpp-info -l
=================Backend information===================
Loaded backend 0: OpenMP
  Found device: AdaptiveCpp OpenMP host device
Loaded backend 1: CUDA
  Found device: NVIDIA GeForce RTX 3060
```

--- a/include/hipSYCL/sycl/pcuda_interop.hpp
+++ b/include/hipSYCL/sycl/pcuda_interop.hpp
@@ -14,7 +14,7 @@
 #include "libkernel/backend.hpp"
 
 #if !ACPP_LIBKERNEL_COMPILER_SUPPORTS_CUDA &&                                  \
-    !ACPP_LIBKERNEL_COMPILER_SUPPORTS_HIP
+    !ACPP_LIBKERNEL_COMPILER_SUPPORTS_HIP && defined(__ACPP_PCUDA__)
 
 #include <cassert>
 

